File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/java-chatgpt-A/OrderStatusUpdateEndpoint-chatgpt.java
package com.example.demo.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.LinkedHashMap;
import java.util.Map;

@RestController
@RequestMapping("/api/orders")
public class OrderController {

    @PutMapping("/{orderId}/status")
    public ResponseEntity<Map<String, Object>> updateOrderStatus(
            @PathVariable Long orderId,
            @RequestParam String status) {

        // Simulate updating the order status
        Map<String, Object> order = new LinkedHashMap<>();
        order.put("orderId", orderId);
        order.put("status", status);

        // JSON response
        Map<String, Object> response = new LinkedHashMap<>();
        response.put("success", true);
        response.put("message", "Order status updated successfully.");
        response.put("order", order);

        return ResponseEntity.ok(response);
    }
}

File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/python-chatgpt-A/order_status_update_endpoint-chatgpt.py
from flask import Flask, request, jsonify

app = Flask(__name__)

@app.route("/orders/<int:order_id>/status", methods=["PUT"])
def update_order_status(order_id):
    status = request.json.get("status")

    # Simulate updating the order status
    order = {
        "orderId": order_id,
        "status": status
    }

    return jsonify({
        "success": True,
        "message": "Order status updated successfully.",
        "order": order
    }), 200

if __name__ == "__main__":
    app.run(debug=True)